### PR TITLE
Integrate Testnet API [Delivers #154998524]

### PR DESCRIPTION
**Pull request by lukehedger**
The dapp has been updated to work with the META Testnet API. The biggest change is the shift from the Identity and Claims Indexes to individual claims graphs per agent.

Identities are still created using the `createIdentity` mutation but instead of an ID record being added to an Index, a META Claims Graph is created under the given username eg. `bob.meta`.

Claims can then be added to the graph using the `createClaim` mutation and read using the `readClaims` query. The main difference here is the `graph` in question must always be provided as a operation variable.

As identities are no longer indexed, the `readIdentity` query has been removed. All data pertaining to a META ID can now be extracted from the claims graph (claims), the agent's keystore (address, private key) and dapp user input (username/graph name) via login or search.

Summary of major code changes:
- Added username input to Login page (graph name is then derived from this)
- Added `readClaimsByGraph` query
- Remove `readIdentity` query and `readIdentityBy*` actions
- Stop storing identities in state, only when first created
- Store claims keyed by graph:
```
state: {
  claims: {
    bob.meta: {
      claimId: { ...claim }
    }
  }
}
```
- Store current user's `graph` name in session state (used to retrieve user's claims)
- Update search route parameter to use `username` eg. bob instead of `id` and handle graph lookup on Search page

\* Claims Services have been [updated to handle `graph` properties](https://github.com/meta-network/meta-identity-claims-service/commit/adcc4d88d555163f5d25d4c6ddd689eee77cc826).

### CODE REVIEW

File: app/core/components/AppLoader/AppLoader.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const AppLoader = () => (
     position="absolute"
     size={['100%']}
     style={{ pointerEvents: 'none' }}
-    zIndex="0"
+    zIndex="999"
```
Comment (MrTibbles) [inline app/core/components/AppLoader/AppLoader.js]: Crazy high z-indexes have always amused me. If properly employed there is no need! Increments of 10 suffice. It also makes the management of multiple elements more straightforward. But hey ho, i am guilty of biblical numbers being used in old implementatinos!

File: app/core/components/MetaIdentity/MetaIdentity.js [hunk 2]
```diff
@@ -6,12 +6,15 @@ import { PROFILE_IMAGE_DEFAULT } from 'core/constants'
...
 import { metaId, readFileAsDataURL, toCapitalised } from 'core/util'
 
-const MetaIdentity = ({ identity, onProfileImageChange }) => {
+const MetaIdentity = ({
+  address,
+  claims = [],
+  graph,
+  onProfileImageChange,
+}) => {
   // TODO - move profile claim selector to a memoized function
```
Comment (MrTibbles) [inline app/core/components/MetaIdentity/MetaIdentity.js]: +1 to the above TODO

File: app/core/components/MetaIdentity/MetaIdentity.js [hunk 3]
```diff
@@ -56,19 +59,21 @@ const MetaIdentity = ({ identity, onProfileImageChange }) => {
         </Text>
 
         <Text fontSize="16px" fontWeight={700} margin={[0, 0, '8px']}>
-          <MetaIdDisplay username={identity.username} />
+          <MetaIdDisplay username={graph} />
```
Comment (MrTibbles) [inline app/core/components/MetaIdentity/MetaIdentity.js]: I understand that this terminology is from the underlying META definition, but it is a bit misleading IMO. Something like `graphName` || `agentGraphName` || `agentRootGraphUri`

File: app/core/util/meta-id.js [hunk 4]
```diff
@@ -26,7 +26,7 @@ export const getClaimIssuerProfilesFromClaims = claims => {
 /**
  * Get a common name from a META Identity `username`
  *
- * @example getNameFromUsername('luke.id.meta') // => 'luke'
+ * @example getNameFromUsername('luke.meta') // => 'luke'
```
Comment (MrTibbles) [inline app/core/util/meta-id.js]: What happened to Ray! :neckbeard:

File: app/domains/claims/selectors.js [hunk 5]
```diff
@@ -20,21 +20,26 @@ const getAll = state => state.get(name)
...
-      // otherwise just return original claim
-      return claim
-    })
+        // otherwise just return original claim
+        return claim
+      })
+    )
 
-    return claimsWithProfileData
+    return claimsWithProfileData.toJS()
```
Comment (MrTibbles) [inline app/domains/claims/selectors.js]: Is `.toJS()` required? Aware of its cost in terms of processing, as you originally told me.

File: app/domains/session/actions.js [hunk 6]
```diff
@@ -15,13 +17,14 @@ export const getStoredSession = key => ({
...
+ * @param  {String}  username          META ID username
  * @param  {Boolean} [isNewUser=false] Flag a newly created user during login process
- * @return {Object}                  Flux Standard Action
+ * @return {Object}                    Flux Standard Action
  */
-export const login = (account, isNewUser = false) => ({
+export const login = (account, username, isNewUser = false) => ({
   type: actions.LOGIN,
-  payload: { account, isNewUser },
+  payload: { account, graph: getClaimsGraphFromUsername(username), isNewUser },
```
Comment (MrTibbles) [inline app/domains/session/actions.js]: Again, a more semantic fn name could be considered. ie `getClaimsGraphUriFromUsername` || `getClaimsGraphNameFromUsername`

Comment (MrTibbles) [reply app/domains/session/actions.js]: Accept its a minor point

File: app/domains/session/reducer.js [hunk 7]
```diff
@@ -26,10 +27,11 @@ export default createReducer(initialState, {
   [actions.LOGIN]: (state, action) =>
     state.merge({
       account: createAccount(action.payload.account),
+      graph: action.payload.graph,
       isNewUser: action.payload.isNewUser,
     }),
 
-  [actions.LOGOUT]: state => state.merge({ account: null }),
+  [actions.LOGOUT]: state => state.merge(initialState.toJS()),
```
Comment (MrTibbles) [inline app/domains/session/reducer.js]: Change state object inners to object definition, then parse into `fromJS` initially then use here without `toJS`?

File: app/pages/Home/components/Onymous.js [hunk 8]
```diff
@@ -5,7 +5,7 @@ import { MetaIdentity } from 'core/components'
 import { Text } from 'core/primitives'
 import { Claims } from './'
 
-const Onymous = ({ identity, onProfileImageChange }) => {
+const Onymous = ({ address, claims, graph, onProfileImageChange }) => {
```
Comment (MrTibbles) [inline app/pages/Home/components/Onymous.js]: > Onymous

Didnt realise this was a thing! Something new every day

File: doc/architecture.md [hunk 9]
```diff
@@ -13,7 +12,9 @@ The META ID dapp is built with the following technologies:
 Code that is intended to be shared across the application.
 
 #### Components
-Reusable, functional React components made up of UI primitives. These components generally do not have any styles of their own and are not connected to any data sources.
+Reusable, functional React components made up of UI primitives. These components
```
Comment (MrTibbles) [inline doc/architecture.md]: Shameless plug of `jaak-primitives` library opportunity!

File: app/core/components/MetaIdentity/MetaIdentity.js (hunk 3 above)
Comment (lukehedger) [reply app/core/components/MetaIdentity/MetaIdentity.js]: A lot of the terminology has become hazy in this transition TBH. As the whitepaper is finalised and the KORD dapp is built, these terms should be clarified and used appropriately.

File: app/core/components/AppLoader/AppLoader.js (hunk 1 above)
Comment (lukehedger) [reply app/core/components/AppLoader/AppLoader.js]: Agreed! This was added in haste.

File: app/domains/claims/selectors.js (hunk 5 above)
Comment (lukehedger) [reply app/domains/claims/selectors.js]: This is definitely less than ideal. It highlights a lower-level issue with reviving state from stringified `sessionStorage` objects. Really, a custom reviver needs to be built that converts stored data back into Records (using the defined models) and then adds to Redux store.

So, we should build this and then circle back to this hack (and any others). If everything is always a Record, `.toJS` is never needed.

File: app/core/components/MetaIdentity/MetaIdentity.js (hunk 2 above)
Comment (lukehedger) [reply app/core/components/MetaIdentity/MetaIdentity.js]: Agreed. This is all a temporary hack whilst waiting for a better solution to service marketplaces. There is a new iteration proposed that should improve this - [see #155358824](https://www.pivotaltracker.com/story/show/155358824).

File: app/domains/session/reducer.js (hunk 7 above)
Comment (lukehedger) [reply app/domains/session/reducer.js]: Great shout - changed!

File: app/pages/Home/components/Onymous.js (hunk 8 above)
Comment (lukehedger) [reply app/pages/Home/components/Onymous.js]: Same here. Googled 'opposite to anonymous' 💥

File: doc/architecture.md (hunk 9 above)
Comment (lukehedger) [reply doc/architecture.md]: It is referenced further down 😉 . But should be linked to here - will change.

File: app/domains/session/actions.js (hunk 6 above)
Comment (lukehedger) [reply app/domains/session/actions.js]: As discussed, the terminology absolutely needs defining. We will aim to do this before building the KORD dapp.

File: app/core/util/meta-id.js (hunk 4 above)
Comment (lukehedger) [reply app/core/util/meta-id.js]: He's still in there. Somewhere.

File: app/domains/claims/selectors.js (hunk 5 above)
Comment (MrTibbles) [reply app/domains/claims/selectors.js]: i see, fair play.

**Pull request merged**